### PR TITLE
Add JS page type meta-docs

### PR DESCRIPTION
**Pull request by wbamberg**
Part of https://github.com/mdn/content/issues/16156.

### CODE REVIEW

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md [hunk 1]
```diff
@@ -68,3 +68,23 @@ This section lists `page-type` values for pages under [Web/CSS](/en-US/docs/Web/
...
+
+- `javascript-class`: a definition of a built-in object, like [`Array`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array).
+- `javascript-constructor`: an object constructor, like [`Array()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array).
+- `javascript-error`: an error, like [RangeError: invalid array length](/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length).
+- `javascript-function`: a built-in function that isn't an object method, like [`encodeuri()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI).
+- `javascript-global-property`: a global property like [`NaN`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN).
+- `javascript-instance-accessor-property`: an accessor property on an object instance, like [`Map.prototype.size`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/size).
+- `javascript-instance-data-property`: a data property on an object instance, like [`Array.prototype.length`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length).
```
Comment (Josh-Cena) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md:82]:
```suggestion
- `javascript-instance-data-property`: a data property on an object instance, like `Array`'s [`length`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length).
```

(I suggest making it correct, in the light of https://github.com/mdn/content/pull/22881)

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md [hunk 2]
```diff
@@ -68,3 +68,23 @@ This section lists `page-type` values for pages under [Web/CSS](/en-US/docs/Web/
...
+
+- `javascript-class`: a definition of a built-in object, like [`Array`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array).
+- `javascript-constructor`: an object constructor, like [`Array()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array).
+- `javascript-error`: an error, like [RangeError: invalid array length](/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length).
+- `javascript-function`: a built-in function that isn't an object method, like [`encodeuri()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI).
+- `javascript-global-property`: a global property like [`NaN`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN).
+- `javascript-instance-accessor-property`: an accessor property on an object instance, like [`Map.prototype.size`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/size).
+- `javascript-instance-data-property`: a data property on an object instance, like `Array`'s [`length`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length).
```
Comment (wbamberg) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md:82]:
```suggestion
- `javascript-instance-data-property`: a data property on an object instance, like the [`length`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) property of `Array`.
```

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md [hunk 3]
```diff
@@ -68,3 +68,23 @@ This section lists `page-type` values for pages under [Web/CSS](/en-US/docs/Web/
...
+- `javascript-instance-method`: a method on an object instance, like [`Array.prototype.at()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at).
+- `javascript-language-feature`: a part of JavaScript syntax not fitting into another category, like [rest parameters](/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).
+- `javascript-namespace`: an object that is not instantiable and has only static members, like [`Math`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math).
+- `javascript-operator`: an operator, like [Addition (+)](/en-US/docs/Web/JavaScript/Reference/Operators/Addition).
+- `javascript-statement`: a statement, like [`switch`](/en-US/docs/Web/JavaScript/Reference/Statements/switch).
+- `javascript-static-accessor-property`: a static accessor property, like [`RegExp.lastMatch`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastMatch).
+- `javascript-static-data-property`: a static data property, like [`Math.E`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/E).
+- `javascript-static-method`: a staticx method, like [`Array.from()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from).
```
Comment (Josh-Cena) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md:90]:
```suggestion
- `javascript-static-method`: a static method, like [`Array.from()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from).
```

**Review by Josh-Cena — APPROVED** [review]
Looks great to me, barring one typo :)

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md [hunk 4]
```diff
@@ -68,3 +68,23 @@ This section lists `page-type` values for pages under [Web/CSS](/en-US/docs/Web/
 - `css-selector`: a [basic selector](/en-US/docs/Web/CSS/CSS_Selectors#basic_selectors), like the [class selector](/en-US/docs/Web/CSS/Class_selectors).
 - `css-shorthand-property`: a [shorthand property](/en-US/docs/Web/CSS/Shorthand_properties), like {{cssxref("background")}}.
 - `css-type`: a [data type](/en-US/docs/Web/CSS/CSS_Types), like [`<color>`](/en-US/docs/Web/CSS/color_value).
+
+## JavaScript page types
+
+This section lists `page-type` values for pages under [Web/JavaScript](/en-US/docs/Web/CSS). Every page in that part of the tree must have a `page-type`, and its value must be one of those listed below or one of the generic page type values.
+
+- `javascript-class`: a definition of a built-in object, like [`Array`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array).
```
Comment (Josh-Cena) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/page_type_key/index.md:76]: This sounds handwavy, but I don't see a way to describe it, so let's just keep it as-is.

**Comment by Josh-Cena** [comment]
@wbamberg You can merge this if you think it's ready.

**Pull request merged**